src/Storage/Device/S3.php

```diff
@@ -657,11 +659,17 @@ public function getPartitionTotalSpace(): float
      * Get all files and directories inside a directory.
      *
      * @param  string  $dir Directory to scan
-     * @return string[]
+     * @return array
+     *
+     * @throws Exception
      */
     public function getFiles(string $dir): array
     {
-        throw new Exception('Not implemented.');
+        $data = $this->listObjects($dir);
+        $data['IsTruncated'] = $data['IsTruncated'] === 'true';
+        $data['KeyCount'] = intval($data['KeyCount']);
```

Comment:
If there are more than 1000 items there won't be any way to see them with this function, let's use a loop and make sure we get all the files

Reply:
Same with LOCAL device we do not support pagination at this point.
But we can add a page size parameter? to overwrite the limit 1000 param?

Reply:
Page size won't help here, the max you can fetch at once with S3 is 1000. 

Let's add a loop to fetch all, we can keep fetching until `IsTruncated` is false, passing the `continuationToken` to the listObjects call to get the next page